syntax = "proto3";

service HelloWorldService {
  rpc hello(HelloRequest) returns (HelloResponse) {}

  rpc sendData(DataRecord) returns (DataResponse) {}
}

message HelloRequest {
  string firstname = 1;
  string lastname = 2;
}

message HelloResponse {
  string text = 1;
}


message DataItem {
  int32 id = 1;
  string name = 2;
  double value = 3;
}

message DataRecord {
  repeated DataItem items = 1;   
}


message DataResponse {
  string message = 1;
}